sdk/core/Azure.Core/tests/HttpClientTransportFunctionalTest.cs

```diff
@@ -58,9 +73,93 @@ public void UsesHttpClientTransportOnNetStandart()
         }
 #endif
 
+        private static IEnumerable CertOptions = new[]
+        {
+            new object[] {true, trustedCert, true },
+            new object[] {true, trustedCert, false },
+            new object[] {true, untrustedCert, true },
+            new object[] {true, untrustedCert, false },
+            new object[] {false, new X509Certificate2(), true },
+            new object[] {false, new X509Certificate2(), false },
+        };
+
+        [TestCaseSource(nameof(CertOptions))]
```

Comment:
We need a full blown functional test for this that starts an inprocess HTTP server makes real request etc. similar to https://github.com/christothes/azure-sdk-for-net/blob/4aa5ff4551243a50560e0d1f06726ad6caec5865/sdk/core/Azure.Core/tests/TransportFunctionalTests.cs